memgpt/openai_tools.py

```diff
@@ -95,32 +207,55 @@ def chat_completion_with_backoff(
     printd(f"Using model {agent_config.model_endpoint_type}, endpoint: {agent_config.model_endpoint}")
     if agent_config.model_endpoint_type == "openai":
         # openai
-        openai.api_base = agent_config.model_endpoint
-        return openai.ChatCompletion.create(
-            model=agent_config.model, messages=messages, functions=functions, function_call=function_call, user=config.anon_clientid
+        # openai.api_base = agent_config.model_endpoint
+        # return openai.ChatCompletion.create(
+        # model=agent_config.model, messages=messages, functions=functions, function_call=function_call, user=config.anon_clientid
+        # )
+        return openai_chat_completions_request(
+            url=agent_config.model_endpoint,  # https://api.openai.com/v1 -> https://api.openai.com/v1/chat/completions
+            api_key=config.openai_key,  # 'sk....'
+            data=dict(
+                model=agent_config.model,
+                messages=messages,
+                functions=functions,
+                function_call=function_call,
+                user=config.anon_clientid,
+            ),
         )
     elif agent_config.model_endpoint_type == "azure":
-        # azure
-        openai.api_type = "azure"
-        openai.api_key = config.azure_key
-        openai.api_base = config.azure_endpoint
-        openai.api_version = config.azure_version
-        if config.azure_deployment is not None:
-            deployment_id = config.azure_deployment
-            engine = None
-            model = config.model
-        else:
-            engine = MODEL_TO_AZURE_ENGINE[config.model]
-            model = None
-            deployment_id = None
-        return openai.ChatCompletion.create(
-            model=model,
-            messages=messages,
-            engine=engine,
-            deployment_id=deployment_id,
-            functions=functions,
-            function_call=function_call,
-            user=client_id,
+        # # azure
+        # openai.api_type = "azure"
+        # openai.api_key = config.azure_key
+        # openai.api_base = config.azure_endpoint
+        # openai.api_version = config.azure_version
+        # if config.azure_deployment is not None:
+        #     deployment_id = config.azure_deployment
+        #     engine = None
+        #     model = config.model
+        # else:
+        #     engine = MODEL_TO_AZURE_ENGINE[config.model]
+        #     model = None
+        #     deployment_id = None
+        # return openai.ChatCompletion.create(
+        #     model=model,
+        #     messages=messages,
+        #     engine=engine,
+        #     deployment_id=deployment_id,
+        #     functions=functions,
+        #     function_call=function_call,
+        #     user=client_id,
+        # )
+        return azure_openai_chat_completions_request(

```

Comment:
@sarahwooders This is the Azure part I'm not sure about

The Azure docs only have a few variables they want when constructing the url for the POST request, and a few of the variables I'm seeing here I don't recognize.

Reply:
https://learn.microsoft.com/en-us/azure/ai-services/openai/reference#embeddings